src/DataGrid.tsx

```diff
@@ -494,8 +494,8 @@ function DataGrid<R, SR, K extends Key>(
 
   function handleKeyDown(event: React.KeyboardEvent<HTMLDivElement>, isEditorPortalEvent = false) {
     if (!(event.target instanceof Element)) return;
-    const isCellEvent = event.target.closest(`.${cellClassname}`) !== null;
-    const isRowEvent = hasGroups && event.target.classList.contains(rowClassname);
+    const isCellEvent = event.target.closest(`.${viewportCellClassname}`) !== null;
+    const isRowEvent = hasGroups && event.target.classList.contains(viewportRowClassname);
```

Comment:
I wonder if this would have been enough 🤔 
```suggestion
    const isCellEvent = event.target.closest('.rdg-row:not(.rdg-summary-row) > .rdg-cell') !== null;
    const isRowEvent = hasGroups && event.target.matches('.rdg-row:not(.rdg-summary-row)');
```

Reply:
This makes sense, the selector can be tweaked in https://github.com/adazzle/react-data-grid/pull/2557